pandas/tests/computation/test_eval.py

```diff
@@ -774,16 +774,19 @@ def check_chained_cmp_op(self, lhs, cmp1, mid, cmp2, rhs):
 f = lambda *args, **kwargs: np.random.randn()
 
 
-ENGINES_PARSERS = list(product(_engines, expr._parsers))
-
 #-------------------------------------
 # typecasting rules consistency with python
 # issue #12388
 
 
 class TestTypeCasting(object):
-
-    def check_binop_typecasting(self, engine, parser, op, dt):
+    @pytest.mark.parametrize('engine', _engines)
+    @pytest.mark.parametrize('parser', expr._parsers)
```

Comment:
Instead of repeating `@pytest.mark.parametrize('parser', expr._parsers)`, you could have a single fixture in the module, and pytest will push those through to each testing function:
```python
@pytest.fixture(params=_engines)
def engine(request):
    return request.param
```